### ISSUE TITLE
Matchers/Filters Not Working

### ISSUE THREAD

**Issue by aress31** [ffuf/ffuf#556]
I tried fuzzing a simple POST request as follows:

```powershell
ffuf.exe -c -v `
  -x 'http://127.0.0.1:8080' `
  -w 'REDACTED\words.lst:WORD' `
  -w 'REDACTED\numbers.lst:NUMBER' `
  -request 'REDACTED\req.txt' `
  -o results.html -of html `
  -mr "SUCCESS"
```
![image](https://user-images.githubusercontent.com/11601622/169676741-81b4f467-a77d-4e48-b765-854c1257361b.png)

With a **successful response** looking like:

```xml
HTTP/2 200 OK
Date: Sun, 22 May 2022 03:11:41 GMT
Content-Type: application/json; charset=utf-8
--- REDACTED ---

{"code":"SUCCESS","msg":"success","data":--- REDACTED---}
```

> Note that this output is directly taken from `burp`.

However, for some obscure reasons `fuff` does not seem to match the successful responses; has this got anything to do with `JSON` and the `regexp`?

**Comment by bsysop** [comment]
Please, send the same test but now remove the `Content-Encoding: gzip` header.
Works?

**Comment by m3ssap0** [comment]
I'm experiencing the same problem for GET requests (`v1.5.0-dev`).

I'm trying to filter out responses with empty `data` object.

Command:
```
./ffuf -request request.txt -w wordlist.txt -fr '"data":\{\}' -t 5 -p 1.0-1.5 -x http://localhost:8080
```

Request in the `request.txt` file:
```
GET /[redacted]/[redacted]?[redacted]=FUZZ HTTP/2
Host: [redacted]
[redacted_custom_headers]


```

Response format (taken from Burp):
```
HTTP/2 200 OK
Content-Type: application/json; charset=utf-8
[redacted]

{"[redacted]":{"status":200},"data":{}}
```

I've also tried different and "simpler" regexes and using matchers instead of filters, but the behavior is the same: it seems that regexes are ignored.

**Comment by joohoi** [comment]
This seems strange indeed. Just in order to make sure Burp isn't doing output formatting in the UI, could you rerun the test with ffuf using `-od somedir` to save the output to a file. The example you provided above definitely should  filter it out. Also note the comment by bsysop above regarding the gzip encoding.

**Comment by m3ssap0** [comment]
I did some tests and I found a workaround reading #526: I activated the rule on Burp to remove the `Accept-Encoding` header.

So it seems that the `Accept-Encoding` header in the request is breaking the functionality.

**Comment by joohoi** [comment]
Yeah, the current version does not uncompress the response body. This will change in the next release with some of the added functionality.
